src/components/UserMenu/UserMenu.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+import { Flex, FlexItem } from '@patternfly/react-core/dist/esm/layouts/Flex';
+import { Avatar } from '@patternfly/react-core/dist/dynamic/components/Avatar';
+import { Dropdown, DropdownItem, DropdownList } from '@patternfly/react-core/dist/esm/components/Dropdown';
+import { Divider } from '@patternfly/react-core/dist/esm/components/Divider';
+import { MenuToggle, MenuToggleElement } from '@patternfly/react-core/dist/esm/components/MenuToggle';
+import { CaretDownIcon } from '@patternfly/react-icons/dist/dynamic/icons/caret-down-icon';
+import ThemePreference from './ThemePreference';
+import { signOut } from 'next-auth/react';
+
+const UserMenu: React.FC = () => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const onSelect = () => {
+    setIsOpen(false);
+  };
+
+  return (
+    <Dropdown
+      isOpen={isOpen}
+      onSelect={onSelect}
+      onOpenChange={(isOpen: boolean) => setIsOpen(isOpen)}
+      toggle={(toggleRef: React.Ref<MenuToggleElement>) => (
+        <MenuToggle ref={toggleRef} aria-label="user menu dropdown" variant="plain" onClick={onToggleClick} isExpanded={isOpen}>
+          <Flex spaceItems={{ default: 'spaceItemsSm' }} style={{ display: 'flex', alignItems: 'center' }}>
+            <FlexItem>
```

Comment:
@aevo98765 Overall the PR looks good, but once i login through the github, it doesn't show the github username or the avatar, it shows the one that is hardcoded here. I think you need to pull the session and use the session.user.name and session.user.image. I locally did the following changes and it worked for me. here is the patch file
https://gist.github.com/vishnoianil/240d67469823365435f128c33717520c

Reply:
Done